src/controllers/BaseController.ts

```diff
@@ -191,18 +193,18 @@ export class BaseController<FORM> extends BaseAsyncHttpController {
                     }
                 }
 
-                const session = request.session.extract();
-                if (session != null) {
-                    const applicationData: ApplicationData = session.getExtraData()
-                        .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
-                        .orDefaultLazy(() => {
-                            const value = {} as ApplicationData;
-                            session.saveExtraData(APPLICATION_DATA_KEY, value);
-                            return value;
-                        });
+                const session: Session | undefined = request.session;
 
-                    // tslint:disable-next-line: max-line-length
-                    applicationData.appeal = that.prepareSessionModelPriorSave(applicationData.appeal || {}, request.body);
+                if (session) {
+                    let applicationData: ApplicationData | undefined = session.getExtraData(APPLICATION_DATA_KEY);
+
+                    if (!applicationData) {
+                        applicationData = {} as ApplicationData;
+                        session.setExtraData(APPLICATION_DATA_KEY, applicationData);
+                    }
+
```

Comment:
this could be simplified
```suggestion
                    session.setExtraData(APPLICATION_DATA_KEY, applicationData || {});
```

Reply:
Good idea, I guess in this case we could actually do this: 

```suggestion
const applicationData: ApplicationData = session.getExtraData(APPLICATION_DATA_KEY) || {} as ApplicationData;
session.setExtraData(APPLICATION_DATA_KEY, applicationData);
```